ctapipe/image/cleaning.py

```diff
@@ -93,3 +93,42 @@ def dilate(geom, mask):
         input mask (array of booleans) to be dilated
     """
     return mask | geom.neighbor_matrix_sparse.dot(mask)
+
+
+def number_of_islands(geom, mask):
+    """
+    Calculate and return the number of connected clusters of a pixel mask given 
+    the camera geometry. 
+    This can be used to seperate between gamma and hadronic showers.
+
+    Parameters
+    ----------
+    geom: `~ctapipe.instrument.CameraGeometry`
+        Camera geometry information
+    mask: ndarray
+        input mask (array of booleans)
+    """
+    hits = np.where(mask)[0]
+    # store information about marked pixels to avoid double counting
+    marked = np.ones(hits.shape[0]) * -1 
```

Comment:
this could be a python `set` or a boolean mask of length `n_pixels`. I don't think an integer array with the indices is the best choice here.